### ISSUE TITLE
NDArithmeticMixin uncertainty propagation for multiplication/division

### ISSUE THREAD

**Issue by MSeifert04** [astropy/astropy#4152]
I used the arithmeticMixin sometime now but only recently I realized that in some circumstances it makes the wrong "decisions". When multiplying something with an uncertainty with something that has no uncertainty it does not even want to propagate the uncertainty (why?)

For example
```
import numpy as np
from astropy.nddata import NDData, NDArithmeticMixin, StdDevUncertainty
class Test(NDArithmeticMixin,NDData):
	pass
uncert = StdDevUncertainty([1,1,1])
a = Test([1,2,3], uncertainty=uncert)
b = Test([1,2,3])
a.multiply(b).uncertainty.array
#Output: array([1, 1, 1])
#but it should be array([1,2,3]) since the uncertainty for 
#a constant times an uncertain value is just the constant times the uncertainty
```
The problem is in https://github.com/astropy/astropy/blob/master/astropy/nddata/mixins/ndarithmetic.py#L149 . There it decides that the uncertainty should not be propagated if only self or operand has an uncertainty. Same problem also arises for division. 

This is not easily fixed since for example the StdDevUncertainty (https://github.com/astropy/astropy/blob/master/astropy/nddata/nduncertainty.py#L329) also raises an Error if the other data has no uncertainty. (also there is a lot of duplicate code in each propagate_* method)

**Comment by mhvk** [comment]
I haven't been following this, but it might be good to combine efforts with #3715, to both propagate uncertainties and track covariances.

**Comment by MSeifert04** [comment]
I just wanted to clarify the issue by comparing the above mentioned to
```
c = Test([1,2,3], uncertainty=StdDevUncertainty([0,0,0])
a.multiply(c).uncertainty.array
#This should return array([1,2,3]) (and does when using NDDataArray) but instead thrown an exception
MissingDataAssociationException: Uncertainty is not associated with an NDData object
```
Unfortunatly there seems to be something else is broken too - which I missed because I used NDDataArray (which does not throw exceptions).

**Comment by MSeifert04** [comment]
Just an aside question: Is NDArithmeticMixin meant to be a standalone (with NDData) or is it only meant as a building block for NDDataArray? I just looked at the test cases (``nddata.mixins.tests.test_ndarithmetic.py``) for the mixins and they are all based on the NDDataArray not on the mixins by themselves.

**Comment by mwcraig** [comment]
@MSeifert04 -- I won't have a time to look at this until later this week but my memory is that the NDArithmeticMixin was originally written as a building block...though if we are calling it a mixin it really ought to be useable with plain NDData.

**Comment by MSeifert04** [comment]
closed since this is solved for 1.2 with PR #4272
